### PR TITLE
Reduce go-libp2p and js-libp2p admins

### PR DESCRIPTION
**Pull request by aschmahmann**
### Summary

- Introduces a team for members of Interplanetary Shipyard working on or with libp2p (mostly a subset of the existing w3dt-stewards group)
- Reduces the number of admins on go-libp2p
- Reduces the number of admins on js-libp2p

### Why do you need this?

The number of admin's on both of these repositories is simply too high. As described in earlier github-mgmt permissions reduction permissions are not meant as tokens of credibility, but of utility. Even the number of Admin's here may be too high, but I don't feel like I've overcut given github-mgmt is here to enable escalation.

I don't feel like I've overcut here and I've run this by @achingbrain and @MarcoPolo who are practically speaking the operating admin's of these repos (i.e. if you've used admin permissions and they don't know about it it's a problem). However, if I've overcut it seems fine to merge and re-add relevant permissions later.

### What else do we need to know?

**DRI:** myself

### Reviewer's Checklist

- [ ] It is clear where the request is coming from (if unsure, ask)
- [ ] All the automated checks passed
- [ ] The YAML changes reflect the summary of the request
- [ ] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

**Comment by aschmahmann** [comment]
Adding as a separate comment so people get notified. The following people will have their permissions reduced. If this is a problem (before or after the merge) open a request to modify the permissions.

Below I'm explicitly calling out anyone whose permissions have dropped from admin to lower than maintain on a repo and added some explicit comments for people who I suspect might care about the permissions reduction

- @dhuseby:
  - will have the permission to go-libp2p change from admin to pull
  - will have the permission to js-libp2p change from admin to pull
  - Do you need push access to any of these repos? If so the Repos Go and js-libp2p-dev groups seem appropriate
- @galargh:
  - will lose admin permission to go-libp2p
  - will lose admin permission to js-libp2p
  - What level of access do you need to these repos? Although you are an org-admin so probably not directly impacted.
-@jbenet:
  - will lose admin permission to go-libp2p
  - will lose admin permission to js-libp2p
  - There haven't been any commits here in ages so I suspect this isn't a problem (we're reducing risk surface here)
- @jorropo:
  - will lose admin permission to go-libp2p
  - will lose admin permission to js-libp2p
  - I suspect you might want Repos Go permissions, if so just ping or send a PR
- @kubuxu:
  - will have the permission to go-libp2p change from admin to push
  - will lose admin permission to js-libp2p
- @laurentsenta:
  - will lose admin permission to go-libp2p
  - will lose admin permission to js-libp2p
  -  What level of access do you need to these repos?
- @mxinden:
  - will lose admin permission to go-libp2p
  - will lose admin permission to js-libp2p
- @p-shahi:
  - will have the permission to go-libp2p change from admin to pull
  - will have the permission to js-libp2p change from admin to pull
  - What level of access do you need here? If you just need push access the Repos Go and js-libp2p-dev groups seem appropriate
- @raulk:
  - will have the permission to go-libp2p change from admin to push
  - will have the permission to js-libp2p change from admin to pull
- @stebalien:
  - will have the permission to js-libp2p change from admin to pull

**Review by MarcoPolo — APPROVED** [review]

File: github/libp2p.yml
Comment (aschmahmann) [inline github/libp2p.yml:2125]: @Stebalien do you think it would still be helpful to continue to have admin here? At the moment I have you on since of the existing admin's that are not from Shipyard you're the only one whose been involved in the code at all.

If not can switch to maintain or leave you with push via Repos - Go.

**Review by mxinden — APPROVED** [review]

**Review by achingbrain — APPROVED** [review]

File: github/libp2p.yml
Comment (Stebalien) [reply github/libp2p.yml:2125]: The only non-push thing I do is:

1. Cleanup branches (should be allowed in `Repos - Go`).
2. Delete spam and ban spammers.

Did we ever find a way to allow everyone to do the latter? If so, `Repos - Go` seems reasonable.

Comment (aschmahmann) [reply github/libp2p.yml:2125]: > Delete spam

IIUC it looks like spam comments is doable with just write permissions to the repo https://docs.github.com/en/communities/moderating-comments-and-conversations/managing-disruptive-comments

> ban spammers

This can be done with an org-wide moderator role which you already have by virtue of being in the github-mgmt stewards group https://github.com/libp2p/github-mgmt/blob/989aed4f3864e6a264e62df4463b40978b7c01b4/github/libp2p.yml#L5077

As a result, I will downgrade to Repos Go

**Review by achingbrain — APPROVED** [review]

**Pull request merged**
